enhancements/installer/connected-onprem-assisted-installer.md

```diff
@@ -0,0 +1,303 @@
+
+## Summary
+
+The assisted installer, [previously proposed and
+accepted](connected-assisted-installer.md), includes a service running on
+cloud.redhat.com to which agents connect and wait for orders on how to
+A) provision the host they are running on, and B) join or form a cluster.
+
+This enhancement proposes to enable a user to run that "assistant" service
+on-premise, inside their own network and on their own infrastructure, for
+one-time-use to create a new cluster.
+
+## Motivation
+
+Many users will not want a service that can provision hosts inside their
+network to be run by a third party outside of their network. Reasons can
+include trust, control, and reproducibility. Those users will prefer to run the
+service on-premise.
+
+### Goals
+
+* Users can run the entire assisted install workflow with software that runs inside their network.
+
+### Non-Goals
+
+* Disconnected and air-gapped environments are not included in this enhancement.
+* Adding Nodes on day 2 will be addressed separately.
+* Utilizing a local container registry for mirrored content will be addressed in a future enhancement.
+
+## Proposal
+
+A user will visit cloud.redhat.com to download a live ISO with which they can
+run the assisted installer on-premise. The RHCOS-based ISO will include:
+
+* the user's pull secret
+* a systemd unit file that runs all containers necessary to run the assisted installer
+
+Similar to downloading the current installer, the download will default to the
+latest available version of the assisted installer, RHCOS and OpenShift.
+
+### User Stories [optional]
+
+#### Story 1
+
+I want to install OpenShift while running the assisted installer and the entire
+installation workflow on infrastructure that I control. I do not want a third
+party cloud service to control the provisioning of my hosts from across the
+internet. I don't mind downloading content over the internet during the install
+workflow.
+
+#### Story 2
+
+I want to automate use of the assisted installer and expect reproducible
+results. Utilizing a cloud service does not provide sufficient control or
+guarantees of reproducibility.
+
+#### Story 3
+
+I want to test the assisted installer prior to production use and expect
+reproducible results. Utilizing a cloud service does not provide sufficient
+control or guarantees of reproducibility.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### RHCOS Live ISO
+
+The user will download an RHCOS-based ISO from cloud.redhat.com, and it will
+already have embedded ignition including their pull secret and the systemd unit
+files required to run the assisted installer.
+
+The user will boot the live ISO and connect to it with their web browser. The
+on-premise assisted installer will itself be run with podman inside the live
+ISO.
+
+The user will need to boot the live ISO in a networking environment where either:
+
```

Comment:
The two steps below mentions "live ISO VM". Is the live ISO limited to running inside a VM? Isn't running the live ISO on a baremetal host also supported?

Reply:
Good call. I added a third option that is not a VM.